packages/custom-functions-metadata/src/parseTree.ts

```diff
@@ -1029,50 +1055,84 @@ function hasInvocationParameter(
 
 /**
  * Gets the parameter type of the node
- * @param t TypeNode
+ * @param node TypeNode
  */
-function getParamType(t: ts.TypeNode, extra: IFunctionExtras, enumList: string[]): string {
+function getParamType(node: ts.TypeNode, extra: IFunctionExtras, enumList: string[]): string {
   let type = "any";
-  // Only get type for typescript files.  js files will return any for all types
-  if (t) {
-    let kind = t.kind;
-    const typePosition = getPosition(t);
-    if (ts.isTypeReferenceNode(t) || ts.isArrayTypeNode(t)) {
-      let arrayType: IArrayType = {
-        dimensionality: 0,
-        type: ts.SyntaxKind.AnyKeyword,
-      };
-      if (ts.isTypeReferenceNode(t)) {
-        const array = t as ts.TypeReferenceNode;
-        if (enumList.indexOf(array.typeName.getText()) >= 0) {
-          // Type found in the enumList
-          return type;
-        }
-        if (array.typeName.getText() !== "Array") {
-          extra.errors.push(logError("Invalid type: " + array.typeName.getText(), typePosition));
-          return type;
-        }
-      }
-      arrayType = getArrayDimensionalityAndType(t);
-      kind = arrayType.type;
+  // Only get type for typescript files. js files will return "any" for all types
+  if (!node) {
+    return type;
+  }
+  const typePosition = getPosition(node);
+  
+  // Get the inner type node if it is an array
+  if (typeNodeIsArray(node)) {
+    let arrayType: IArrayType = {
+      dimensionality: 0,
+      node: node,
+    };
+    arrayType = getArrayDimensionalityAndType(node);
+    node = arrayType.node;
+  }
+
+  // We currently accept the following types of reference node: enum will be converted to "any",
+  // Excel.CellValue will be converted accordingly. Anything else is invalid. (Array reference node has already been covered above.)
+  if (ts.isTypeReferenceNode(node)) {
+    const typeReferenceNode = node as ts.TypeReferenceNode;
+    let nodeTypeName = typeReferenceNode.typeName.getText();
+    if (enumList.indexOf(nodeTypeName) >= 0) {
+      // Type found in the enumList
+      return type;
     }
     // @ts-ignore
-    type = TYPE_MAPPINGS[kind];
-    if (!type) {
-      extra.errors.push(logError("Type doesn't match mappings", typePosition));
+    if (CELLVALUETYPE_MAPPINGS[nodeTypeName]) {
+      // @ts-ignore
+      let cellValue = CELLVALUETYPE_MAPPINGS[nodeTypeName]
+      if (cellValue === "unsupported") {
+        extra.errors.push(logError("Custom function does not support cell value type: " + typeReferenceNode.typeName.getText(), typePosition));
+        return type;
+      }
+      return cellValue;
     }
+
+    extra.errors.push(logError("Invalid type: " + typeReferenceNode.typeName.getText(), typePosition));
+    return type;
+  }
+
+  // @ts-ignore
+  type = TYPE_MAPPINGS[node.kind];
+  if (!type) {
+    extra.errors.push(logError("Type doesn't match mappings", typePosition));
```

Comment:
This error message is describing the implementation (using the type mappings array). Please compose an error message meaningful to the user. For example, in ScriptLab, if this error was encountered, what is really the problem to tell the user. Is it that they used a type that isn't supported?

Reply:
Hi, I moved the code around a bit for readability, but this error message was created by someone else. I'm not sure if I understand what this is for, or if the code ever hits, and I prefer not changing it for safety.